pkgs/development/libraries/science/math/petsc/default.nix

```
@@ -9,7 +27,12 @@ stdenv.mkDerivation rec {
     sha256 = "04vy3qyakikslc58qyv8c9qrwlivix3w6znc993i37cvfg99dch9";
   };
 
-  nativeBuildInputs = [ blas gfortran gfortran.cc.lib lapack python ];
+  nativeBuildInputs = [ python gfortran gfortran.cc.lib ];
+  buildInputs = [ blas lapack ]
+    ++ lib.optional mpiSupport mpi
+    ++ lib.optional (mpiSupport && mpi == openmpi) openssh
```

```suggestion
    ++ lib.optional (mpiSupport && mpi.pname == "openmpi") openssh
```
Your version compares paths, which could potentially be different if `mpi` and `openmpi` are different variants of `openmpi` (e.g. due to an overlay).

---

Very good catch!